newIDE/app/src/MainFrame/index.js

```diff
@@ -578,15 +583,48 @@ const MainFrame = (props: Props) => {
       ].includes(kind)
         ? `${kind} ${name}`
         : kind;
+
+      let customIconUrl = '';
+      if (
+        (kind === 'events functions extension' ||
+          kind === 'events functions extension') &&
+        project &&
+        project.hasEventsFunctionsExtensionNamed(name)
+      ) {
+        const eventsFunctionsExtension = project.getEventsFunctionsExtension(
+          name
+        );
+        customIconUrl = eventsFunctionsExtension.getIconUrl();
+      }
       const icon =
-        kind === 'start page' ? <HomeIcon titleAccess="Home" /> : undefined;
+        kind === 'start page' ? (
+          <HomeIcon titleAccess="Home" />
+        ) : kind === 'debugger' ? (
+          <DebuggerIcon />
+        ) : kind === 'external events' ? (
+          <ExternalEventsIcon />
+        ) : kind === 'external layout' ? (
+          <ExternalLayoutIcon />
+        ) : kind === 'events functions extension' ? (
+          <ExtensionIcon />
+        ) : null;
+
       const closable = kind !== 'start page';
       const extraEditorProps =
         kind === 'start page'
           ? { storageProviders: props.storageProviders }
           : undefined;
       return {
         icon,
+        renderCustomIcon: customIconUrl
+          ? (lightness: number) => (
```

Comment:
To me brightness is more understandable that lightness that could relate to weight.